File: 151003/Matoshko/tweet-service/src/main/java/services/tweetservice/domain/entity/Tweet.java
package services.tweetservice.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.proxy.HibernateProxy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@Entity
@Table(name = "tbl_tweet")
public class Tweet extends BaseEntity {
    @ManyToOne
    @JoinColumn(name = "creator_id")
    private Creator creator;
    @Column(name = "title", unique = true, nullable = false)
    private String title;
    @Column(name = "content")
    private String content;
    @CreatedDate
    @Column(name = "created")
    private LocalDateTime created;
    @LastModifiedDate
    @Column(name = "modified")
    private LocalDateTime modified;
    @OneToMany(mappedBy = "tweet")
    private List<Message> messageList = new ArrayList<>();
    @OneToMany(mappedBy = "tweet")
    private List<TweetSticker> tweetStickerList = new ArrayList<>();

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        Class<?> oEffectiveClass = o instanceof HibernateProxy ? ((HibernateProxy) o).getHibernateLazyInitializer().getPersistentClass() : o.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Tweet tweet = (Tweet) o;
        return id != null && Objects.equals(id, tweet.id);
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }
}

File: 151004/Sakovsky/publisher/src/entities/Tweet.ts
import { TweetRequestToCreate } from "src/dto/request/TweetRequestToCreate";
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn, TableForeignKey } from "typeorm";
import { Author } from "./Author";

@Entity({name: 'tbl_tweets'})
export class Tweet {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    @ManyToOne(() => Author, (author) => author.id)
    authorId: number;

    @ManyToOne(() => Author, author => author.id)
    author: Author

    @Column({unique: true})
    title: string;

    @Column()
    content: string;

    @Column({nullable: false, default: new Date()})
    created: Date;

    @Column({nullable: true, default: null})
    modified: Date;

    setTweet(dto: TweetRequestToCreate){
        this.authorId = dto.authorId;
        this.content = dto.content;
        this.title = dto.title;
        this.created = new Date();//new Date().toISOString().slice(0, 19).replace("T", " ")
        this.modified = null;
    }
}